sentence.go

```diff
@@ -110,54 +170,45 @@ func Checksum(s string) string {
 	return fmt.Sprintf("%02X", checksum)
 }
 
+// MustRegisterParser register a custom parser or panic
+func MustRegisterParser(t string, parser ParserCallback) {
+	if err := RegisterParser(t, parser); err != nil {
+		panic(err)
+	}
+}
+
+// RegisterParser register a custom parser
+func RegisterParser(t string, parser ParserCallback) error {
+	if _, ok := customParsers[t]; ok {
+		return fmt.Errorf("nmea: parser for prefix '%s' already exists", t)
+	}
+
+	customParsers[t] = parser
+	return nil
+}
+
 // Parse parses the given string into the correct sentence type.
 func Parse(raw string) (Sentence, error) {
 	s, err := parseSentence(raw)
 	if err != nil {
 		return nil, err
 	}
+
+	// Custom parser allow overriding of existing parsers
+	if customParserCallback, ok := customParsers[s.Type]; ok {
```

Comment:
Call this variable `parser`.

Reply:
I think custom parsers should be forced to match the entire message prefix, talker + type.

Reply:
I disagree, in my experience different talkers can provide the same sentence type. I do see a case where you might want to match on the whole prefix though, perhaps this can be a future task.